### PR TITLE
Support exceptions.

### PR DESCRIPTION
**Pull request by Timandes**
Thanks to @ryanuber 's Pull request #28. 

We need more tests after merging.

### CODE REVIEW

**Comment by andreiz** [comment]
We should probably bump the version significantly, because exceptions will most likely mean people need to modify their code for it to continue functioning.

**Comment by fabiorphp** [comment]
I agree with you @andreiz! 
@Timandes , is very important that you make tests for this PR.

**Comment by fabiorphp** [comment]
@andreiz How we could resolve this issue about version in pecl repository?

**Comment by andreiz** [comment]
We update the version here, then build a PECL tarball and upload it.

**Comment by fabiorphp** [comment]
Awesome @andreiz ! 
Please, don't build the package now, wait @Timandes  make tests.

**Comment by andreiz** [comment]
I'm not building anything until this is merged in.

**Comment by fabiorphp** [comment]
:+1:

**Comment by Timandes** [comment]
@fabiorphp I'm not good at writing test case(maybe because i'm not good at english). Could you guide me to do this?

**Comment by fabiorphp** [comment]
Of course! I'm here to help you.

Before we start, please look at the tests that are already created. The tests are very easy.

**Comment by Timandes** [comment]
OK, i'll start with existing case.
I saw a lot of fails on travis-ci.....

**Comment by fabiorphp** [comment]
It's normal. If you prefer run in your machine, please look the dev-tools folder.

I recommend that you install the Zookeeper server for you create integration tests.

**Comment by fabiorphp** [comment]
Let us wait the @andreiz integrating the repository with Travis-CI and Coveralls, it's more easy to approve pull-requests

**Comment by Timandes** [comment]
@fabiorphp Wait a moment.
I found some compiling errors under PHP with ZTS enabled.

**Comment by Timandes** [comment]
done : )

**Comment by fabiorphp** [comment]
I will test soon.

File: tests/check_if_exists_without_connect.phpt [hunk 1]
```diff
@@ -8,6 +8,17 @@ if (!extension_loaded('zookeeper')) {
 --FILE--
 <?php
 $client = new Zookeeper();
-$client->exists('/test1');
+try {
```
Comment (fabiorphp) [inline tests/check_if_exists_without_connect.phpt]: I think that test must be simple.
E.g:

```php
--TEST--
Should throw exception when check if node exists without connect
--SKIPIF--
<?php
if (!extension_loaded('zookeeper')) {
    echo 'Zookeeper extension is not loaded'
};
--FILE--
<?php
$client = new Zookeeper();
try {
    $client->exists('/test1');
} catch(Exception $e) {
    echo $e->getMessage() . "\n" . $e->getCode();
}
--EXPECT--
Zookeeper->connect() was not called
5998
```

Comment (Timandes) [inline tests/check_if_exists_without_connect.phpt]: Does `--EXPECT--` part support PHP code?

like:

```php
--EXPECT--
Zookeeper->connect() was not called
<?php echo Zookeeper::CONNECT_NOT_CALLED; ?>
```

I'm worried about future changing of constant value.

Is this my over-worriness?

Comment (fabiorphp) [inline tests/check_if_exists_without_connect.phpt]: Please, see these docs bellow and be  a monster in making tests!

https://qa.php.net/write-test.php
https://qa.php.net/phpt_details.php

Comment (Timandes) [inline tests/check_if_exists_without_connect.phpt]: Monster +_+....

**Comment by Timandes** [comment]
Done.

**Comment by fabiorphp** [comment]
Well, @Timandes all tests passed!
Now, we wait @andreiz merge this PR.

@andrei look at the Travis-CI integration guide:
http://docs.travis-ci.com/user/getting-started/

**Comment by fabiorphp** [comment]
Anything new?

**Comment by andreiz** [comment]
Sorry, I'll try to do this today.

**Comment by fabiorphp** [comment]
Awesome @andreiz :+1:

**Comment by andreiz** [comment]
So what should go into .travis.yml file, just this?

language: php
php:
  - 5.4
  - 5.5
  - 5.6

**Comment by fabiorphp** [comment]
The Travis file was created and is on the master branch:
https://github.com/andreiz/php-zookeeper/blob/master/.travis.yml.

You should only enable for this repo.

**Comment by andreiz** [comment]
I enabled the TravisCI integration and kicked off a build.

**Comment by fabiorphp** [comment]
Awesome! :+1:

**Comment by fabiorphp** [comment]
@Timandes Can you make push --force for run Travis-CI integration?

**Comment by Timandes** [comment]
@fabiorphp How should i do ... ?

**Comment by fabiorphp** [comment]
It is simple!
Use commit ammend and after that, push force

**Comment by Timandes** [comment]
```
$ git push --force
warning: push.default is unset; its implicit value is changing in
Git 2.0 from 'matching' to 'simple'. To squelch this message
and maintain the current behavior after the default changes, use:

  git config --global push.default matching

To squelch this message and adopt the new behavior now, use:

  git config --global push.default simple

See 'git help config' and search for 'push.default' for further information.
(the 'simple' mode was introduced in Git 1.7.11. Use the similar mode
'current' instead of 'simple' if you sometimes use older versions of Git)

Username for 'https://github.com': Timandes
Password for 'https://Timandes@github.com':
Everything up-to-date
```

I'm not sure that i did a right thing ...

**Comment by fabiorphp** [comment]
Well, is git push push -f origin master in you repo

**Comment by Timandes** [comment]
Is there anything else i can do?

**Comment by fabiorphp** [comment]
Not know. We will wait the @andreiz decide.

**Comment by Timandes** [comment]
You never sleep? ...
Why do i always see you online whenever i'm online?...

**Comment by fabiorphp** [comment]
Never sleep. Sleep is for the weak.

**Comment by Timandes** [comment]
洗洗睡吧……

File: php_zookeeper.h [hunk 2]
```diff
@@ -69,6 +69,11 @@ PS_FUNCS(zookeeper);
 
 #endif /* HAVE_ZOOKEEPER_SESSION */
 
+#ifndef PHPZK_ADDITIONAL_CONSTANTS
```
Comment (andreiz) [inline php_zookeeper.h]: Where is PHPZK_ADDITIONAL_CONSTANTS ever defined?

Comment (Timandes) [inline php_zookeeper.h]: Maybe it's useless.
Should I remove it?

**Comment by Timandes** [comment]
```
--2015-06-30 01:10:19--  (try:20)  http://apache.fayea.com/zookeeper/zookeeper-3.4.6/zookeeper-3.4.6.tar.gz
Connecting to apache.fayea.com (apache.fayea.com)|119.6.56.18|:80... failed: Connection timed out.
Giving up.
```

**Comment by Timandes** [comment]
Is there anything more I can do?

**Comment by fabiorphp** [comment]
Something new guys?

**Comment by Timandes** [comment]
Hope so.

**Comment by andreiz** [comment]
Is this ready to be merged then?

**Comment by Timandes** [comment]
:ok_hand:

**Comment by fabiorphp** [comment]
Yep!

**Comment by fabiorphp** [comment]
Something new guys?

**Comment by fabiorphp** [comment]
Something new guys? Or that PR will stand still as the other PR #28

**Comment by fabiorphp** [comment]
Can I merge this request? I have permission to do that.

**Comment by andreiz** [comment]
I'll bump the version and make a release.

**Pull request merged**